opm/autodiff/WellHelpers.hpp

```diff
@@ -191,6 +191,63 @@ namespace Opm {
             return retval;
         }
 
+        template <class MatrixType>
+        bool invertMatrixLapack(MatrixType& matrix) {
+
+            const int m = matrix.M();
+            const int n = matrix.N();
+            assert(m == n); // it needs to be a square matrix
+
+            // the matrix
+            double* A = new double[m * n];
+
+            // copying the data in matrix to A
+            int i_count = 0;
+            for (int irow = 0; irow < m; ++irow) {
+                for (int icol = 0; icol < n; ++icol) {
+                    A[i_count] = matrix[irow][icol];
+                    ++i_count;
+                }
+            }
+
+            int* ipiv = new int[n + 1];
+            int lwork = m * n;
+            double* work = new double[lwork];
+            int info;
+
+            dgetrf_(&m, &n, A, &n, ipiv, &info);
+
+            if (info != 0) {
+                delete [] ipiv;
+                delete [] work;
+                delete [] A;
+                return false;
+            }
+
+            dgetri_(&m, A, &n, ipiv, work, &lwork, &info);
+
+            if (info != 0) {
+                delete [] ipiv;
+                delete [] work;
+                delete [] A;
+                return false;
+            }
+
+            // copying inverted A back to matrix
+            i_count = 0;
+            for (int irow = 0; irow < m; ++irow) {
+                for (int icol = 0; icol < n; ++icol) {
+                    matrix[irow][icol] = A[i_count];
```

Comment:
This loop nest also assumes that the matrix is symmetric.  Use same rewrite as above.